src/dbnode/storage/series/lookup/entry.go

```diff
@@ -133,6 +136,17 @@ func (entry *Entry) IndexedForBlockStart(indexBlockStart xtime.UnixNano) bool {
 	return isIndexed
 }
 
+// RequiresColdFlushForBlockStart returns a bool to indicate
+// if the Entry was written as a cold write or not.
+func (entry *Entry) RequiresColdFlushForBlockStart(blockStart xtime.UnixNano) bool {
+	v, ok := entry.queryableBlockRetriever.BlockStatesSnapshot().UnwrapValue()
+	if !ok {
+		return false
+	}
+	coldBlocks := entry.Series.ColdFlushBlockStarts(v)
```

Comment:
This method unfortunately doesn't quite do what I think you might think it does, it only returns cold blocks that "need to be flushed":
https://github.com/m3db/m3/blob/2e48cd9759e228a92114eaf2b20d28e7f1ee19aa/src/dbnode/storage/series/buffer.go#L436:L445

I would go and implement a new method on `src/dbnode/storage/series/series.go` which calls `src/dbnode/storage/series/buffer.go` and returns True if a block start has a cold bucket that's not empty.

i.e. something like:
```go
func (b *dbBuffer) ColdWritesAtBlockStartExist(blockStart xtime.UnixNano) bool {
	bucketVersions, ok := b.bucketsMap[blockStart]
	if !ok {
		return false
	}

	for _, bucket := range bucketVersions.buckets {
		if bucket.writeType == ColdWrite && bucket.streamsLen() > 0 {
			return true
		}
	}

	return false
}
```